File: jeffrey-local/profiles/profile-management/src/main/java/pbouda/jeffrey/profile/manager/FlagsData.java
/*
 * Jeffrey
 * Copyright (C) 2025 Petr Bouda
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package pbouda.jeffrey.profile.manager;

import pbouda.jeffrey.provider.profile.model.JvmFlagDetail;

import java.util.List;
import java.util.Map;

/**
 * Response data for the JVM Flags dashboard.
 *
 * @param flagsByOrigin Map of flags grouped by origin (Default, Ergonomic, Command line, Management)
 * @param totalFlags    Total number of flags
 * @param changedFlags  Number of flags whose values changed during the recording
 */
public record FlagsData(
        Map<String, List<JvmFlagDetail>> flagsByOrigin,
        int totalFlags,
        int changedFlags
) {
}

File: jeffrey-microscope/pages-microscope/src/services/api/model/FlagsData.ts
/*
 * Jeffrey
 * Copyright (C) 2025 Petr Bouda
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import JvmFlag from './JvmFlag';

/**
 * Response data for the JVM Flags dashboard.
 */
export default interface FlagsData {
  /** Map of flags grouped by origin (Default, Ergonomic, Command line, Management) */
  flagsByOrigin: Record<string, JvmFlag[]>;
  /** Total number of flags */
  totalFlags: number;
  /** Number of flags whose values changed during the recording */
  changedFlags: number;
}
